dspace-api/src/main/java/org/dspace/authorize/dao/impl/ResourcePolicyDAOImpl.java

```diff
@@ -36,79 +38,99 @@ protected ResourcePolicyDAOImpl() {
-        criteria.add(Restrictions.and(
-            Restrictions.eq("dSpaceObject", dso),
-            Restrictions.eq("epersonGroup", group),
-            Restrictions.eq("actionId", action)
-        ));
-        criteria.add(Restrictions.and(Restrictions.not(Restrictions.eq("id", notPolicyID))));
-        return list(criteria);
+    public List<ResourcePolicy> findByTypeGroupActionExceptId(Context context, DSpaceObject dso, Group group, int
+        action, int notPolicyID) throws SQLException {
+        CriteriaBuilder criteriaBuilder = getCriteriaBuilder(context);
+        CriteriaQuery criteriaQuery = getCriteriaQuery(criteriaBuilder, ResourcePolicy.class);
+        Root<ResourcePolicy> resourcePolicyRoot = criteriaQuery.from(ResourcePolicy.class);
+        criteriaQuery.select(resourcePolicyRoot);
+        criteriaQuery.where(criteriaBuilder.and(criteriaBuilder.equal(resourcePolicyRoot.get(ResourcePolicy_
+                .dSpaceObject), dso),
+            criteriaBuilder.equal(resourcePolicyRoot.get(ResourcePolicy_.epersonGroup), group),
+            criteriaBuilder.equal(resourcePolicyRoot.get(ResourcePolicy_.actionId), action),
+            criteriaBuilder.notEqual(resourcePolicyRoot.get(ResourcePolicy_.id), notPolicyID)
+            )
+        );
+        return list(context, criteriaQuery, false, ResourcePolicy.class, 1, -1);
     }
 
-    public List<ResourcePolicy> findByEPersonGroupTypeIdAction(Context context, EPerson e, List<Group> groups,
-                                                               int action, int type_id) throws SQLException {
-        Criteria criteria = createCriteria(context, ResourcePolicy.class);
-        criteria.add(Restrictions.and(
-            Restrictions.eq("resourceTypeId", type_id),
-            Restrictions.eq("actionId", action),
-            (Restrictions.or(
-                Restrictions.eq("eperson", e),
-                Restrictions.in("epersonGroup", groups)
-            ))
-        ));
-        return list(criteria);
+    public List<ResourcePolicy> findByEPersonGroupTypeIdAction(Context context, EPerson e, List<Group> groups, int
+        action, int type_id) throws SQLException {
+        //TODO Make sure you double check this
+        CriteriaBuilder criteriaBuilder = getCriteriaBuilder(context);
+        CriteriaQuery criteriaQuery = getCriteriaQuery(criteriaBuilder, ResourcePolicy.class);
+        Root<ResourcePolicy> resourcePolicyRoot = criteriaQuery.from(ResourcePolicy.class);
+        criteriaQuery.select(resourcePolicyRoot);
+        criteriaQuery.where(criteriaBuilder.and(criteriaBuilder.equal(resourcePolicyRoot.get(ResourcePolicy_
+                .resourceTypeId), type_id),
+            criteriaBuilder.equal(resourcePolicyRoot.get(ResourcePolicy_.actionId), action),
+            criteriaBuilder.or(criteriaBuilder.equal(resourcePolicyRoot.get(ResourcePolicy_.eperson), e),
+                criteriaBuilder.in(resourcePolicyRoot.get(ResourcePolicy_.epersonGroup).in(groups)))));
```

Comment:
Code alignment here is extremely hard on the eyes. It's very difficult to see where the `and` starts/ends and the `or` starts/ends.  I'd really recommend (throughout this PR) that we not auto-align (which this looks to be using), and instead  align the code in a way that it's easier to see the structure.   Our Code Style supports this idea, but you may need to manually update the alignment in areas.

This isn't a showstopper, but it makes these queries a TON easier to understand.  This current alignment is nearly impossible to grasp.

Reply:
(And this isn't the only place where queries could be more naturally aligned.  Most of them seem to be hard-wrapped in a manner that looks like an IDE simply wrapped it at 120 chars...instead of aligning things in a way that allows for easier reading)